src/cueTextTracks.js

```diff
@@ -0,0 +1,136 @@
+/**
+* This feature allows metadata text tracks to be manipulated once available, @see process.
+* It also allows ad implementations to leverage ad cues coming through
+* text tracks, @see processAdTrack
+**/
+
+import videojs from 'video.js';
+
+const cueTextTracks = {};
+
+/**
+* This feature allows metadata text tracks to be manipulated once they are available,
+* usually after the 'loadstart' event is observed on the player
+* @param player A reference to a player
+* @param processTrack A method that performs some operations on a metadata text track
+**/
+cueTextTracks.process = function(player, processTrack) {
+  const tracks = player.textTracks();
+  const prepareTrack = function(track) {
+    if (track.kind === 'metadata') {
+      cueTextTracks.setTrackMode(track);
+      processTrack(player, track);
+    }
+  };
+
+  // Text tracks are available
+  if (tracks.length > 0) {
```

Comment:
Do you need this condition? Would it be safer to both run the forloop and attach the event handler?